chain/network/src/codec.rs

```diff
@@ -101,6 +152,30 @@ mod test {
         assert_eq!(bytes_to_peer_message(&decoded).unwrap(), msg);
     }
 
+    #[test]
+    fn test_tx_forward() {
+        let pk = PublicKey::empty(KeyType::ED25519);
+        let tx = SignedTransaction::new(
+            Signature::default(),
+            Transaction::new(
+                "tmp".to_string(),
+                pk.clone(),
+                "other".to_string(),
+                0,
+                CryptoHash::default(),
+            ),
+        );
+        let msg = PeerMessage::Routed(RoutedMessage {
+            target: PeerIdOrHash::Hash(CryptoHash::default()),
+            author: PeerId(pk),
+            signature: Default::default(),
+            ttl: 0,
+            body: RoutedMessageBody::ForwardTx(tx),
+        });
+        let bytes = msg.try_to_vec().unwrap();
+        assert!(is_forward_tx(&bytes));
```

Comment:
We need to cover other situations (SECP key, hash instead of peer id, etc).

Reply:
Done in https://github.com/nearprotocol/nearcore/pull/3317/commits/967492795877223427b202b95594944c35f1879a